### PR TITLE
Aarch64 arch

### PR DESCRIPTION
**Pull request by MrXinWang**

### CODE REVIEW

File: arch/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ acpi = ["acpi_tables"]
 
 [dependencies]
 byteorder = "1.3.4"
-kvm-bindings = { git = "https://github.com/cloud-hypervisor/kvm-bindings", branch = "ch" }
+kvm-bindings = { git = "https://github.com/cloud-hypervisor/kvm-bindings", branch = "ch", features = ["fam-wrappers"] }
```
Comment (michael2012z) [inline arch/Cargo.toml]: Do we need it?

File: arch/Cargo.toml [hunk 2]
```diff
@@ -23,3 +23,4 @@ features = ["elf", "bzimage"]
 
 [dev-dependencies]
 rand = "0.7.3"
+device_tree = ">=1.1.0"
```
Comment (michael2012z) [inline arch/Cargo.toml]: Do we need it?

File: arch/src/aarch64/layout.rs [hunk 3]
```diff
@@ -1,7 +1,86 @@
...
+/// Legacy space will be allocated at once whiling setting up legacy devices.
+pub const LEGACY_DEVICES_MAPPED_IO_SIZE: u64 = 0x0700_0000;
+
+/// Starting from 0x1000_0000 (256MiB), the 768MiB (ends at 1 GiB) is used for PCIE MMIO
+pub const PCI_DEVICES_MAPPED_IO_START: u64 = 0x1000_0000;
+pub const PCI_DEVICES_MAPPED_IO_SIZE: u64 = 0x3000_0000;
+
+/// PCI MMCONFIG space (start: after the device space at 1 GiB, length: 256MiB)
+pub const PCI_MMCONFIG_START: GuestAddress = GuestAddress(0x4000_0000);
+pub const PCI_MMCONFIG_SIZE: GuestUsize = (256 << 20);
```
Comment (michael2012z) [inline arch/src/aarch64/layout.rs]: unnecessary parentheses around assigned value

File: arch/src/aarch64/mod.rs [hunk 4]
```diff
@@ -20,20 +28,119 @@ pub struct EntryPoint {
...
-/// Stub function that needs to be implemented when aarch64 functionality is added.
-pub fn get_reserved_mem_addr() -> usize {
-    0
+/// Returns the memory address where the kernel could be loaded.
+pub fn get_kernel_start() -> u64 {
+    layout::RAM_64BIT_START
+}
+
+/// Returns the memory address where the initrd could be loaded.
+pub fn initrd_load_addr(guest_mem: &GuestMemoryMmap, initrd_size: usize) -> super::Result<u64> {
```
Comment (michael2012z) [inline arch/src/aarch64/mod.rs]: We don't need this function, remove it.

File: arch/src/aarch64/regs.rs [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+#[allow(non_upper_case_globals)]
+// PSR (Processor State Register) bits.
+// Taken from arch/arm64/include/uapi/asm/ptrace.h.
+const PSR_MODE_EL1h: u64 = 0x0000_0005;
+const PSR_F_BIT: u64 = 0x0000_0040;
+const PSR_I_BIT: u64 = 0x0000_0080;
+const PSR_A_BIT: u64 = 0x0000_0100;
+const PSR_D_BIT: u64 = 0x0000_0200;
+// Taken from arch/arm64/kvm/inject_fault.c.
+const PSTATE_FAULT_BITS_64: u64 = (PSR_MODE_EL1h | PSR_A_BIT | PSR_F_BIT | PSR_I_BIT | PSR_D_BIT);
```
Comment (michael2012z) [inline arch/src/aarch64/regs.rs]: unnecessary parentheses around assigned value

**Review by michael2012z — CHANGES_REQUESTED** [review]

File: arch/Cargo.toml (hunk 2 above)
Comment (MrXinWang) [reply arch/Cargo.toml]: deleted

File: arch/src/aarch64/regs.rs (hunk 5 above)
Comment (MrXinWang) [reply arch/src/aarch64/regs.rs]: done

File: arch/src/aarch64/layout.rs (hunk 3 above)
Comment (MrXinWang) [reply arch/src/aarch64/layout.rs]: done

File: arch/Cargo.toml (hunk 1 above)
Comment (MrXinWang) [reply arch/Cargo.toml]: Done

**Pull request closed without merging**